From 0e96f3f0a06e5daf126baeb345d23dad3198efc6 Mon Sep 17 00:00:00 2001
From: Patrick Gaskin <patrick@pgaskin.net>
Date: Fri, 1 Jan 2021 09:36:18 -0500
Subject: [PATCH] utils: Fix WSASYSNOTREADY error for pacat and pactl on win32

TODO: It works, but is this a proper fix? I need to look at the
WSAStartup in pulsecore/dllmain.c and figure out why it's there. I'll
also need to see if pipes are used anywhere else on win32 other than
the mainloop on win32.
---
 src/utils/meson.build | 4 ++--
 src/utils/pacat.c     | 7 +++++++
 src/utils/pactl.c     | 7 +++++++
 3 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/src/utils/meson.build b/src/utils/meson.build
index f40eaff95..f57f39e39 100644
--- a/src/utils/meson.build
+++ b/src/utils/meson.build
@@ -15,7 +15,7 @@ executable('pacat',
   install_rpath : privlibdir,
   include_directories : [configinc, topinc],
   link_with : [libpulsecommon, libpulse],
-  dependencies : [sndfile_dep, libintl_dep],
+  dependencies : [sndfile_dep, libintl_dep, platform_socket_dep],
   c_args : pa_c_args,
 )
 
@@ -38,7 +38,7 @@ executable('pactl',
   install_rpath : privlibdir,
   include_directories : [configinc, topinc],
   link_with : [libpulsecommon, libpulse],
-  dependencies : [sndfile_dep, libintl_dep],
+  dependencies : [sndfile_dep, libintl_dep, platform_socket_dep],
   c_args : pa_c_args,
 )
 
diff --git a/src/utils/pacat.c b/src/utils/pacat.c
index 4d2ecf717..d5bb13d0f 100644
--- a/src/utils/pacat.c
+++ b/src/utils/pacat.c
@@ -785,6 +785,13 @@ int main(int argc, char *argv[]) {
     bindtextdomain(GETTEXT_PACKAGE, PULSE_LOCALEDIR);
 #endif
 
+#ifdef OS_IS_WIN32
+    {
+        WSADATA data;
+        WSAStartup(MAKEWORD(2, 0), &data);
+    }
+#endif
+
     bn = pa_path_get_filename(argv[0]);
 
     if (strstr(bn, "play")) {
diff --git a/src/utils/pactl.c b/src/utils/pactl.c
index bc1c5265a..84e2c74e0 100644
--- a/src/utils/pactl.c
+++ b/src/utils/pactl.c
@@ -1740,6 +1740,13 @@ int main(int argc, char *argv[]) {
     bindtextdomain(GETTEXT_PACKAGE, PULSE_LOCALEDIR);
 #endif
 
+#ifdef OS_IS_WIN32
+    {
+        WSADATA data;
+        WSAStartup(MAKEWORD(2, 0), &data);
+    }
+#endif
+
     bn = pa_path_get_filename(argv[0]);
 
     proplist = pa_proplist_new();
-- 
2.29.2

